WordPress/Classes/Models/WPAccount+DeduplicateBlogs.swift

```diff
@@ -22,11 +23,11 @@ extension BlogService {
                 continue
             }
```

Comment:
Sort of out of scope, but I wonder whether the could would be clearer if we did some filtering before attempting the deduplication instead of adding checks in every iteration

```swift
let blogsById = Dictionary(
  grouping: blogs.filter { $0.dotComID != .none },
  by: { $0.dotComID!.intValue }
)

for (blogId, group) in blogsByID where group.count > 1 {
  // no checks, just call deduplicate
}
```

Or... is it best to be defensive, just in case?

Reply:
If we want to _exactly_ match the original logic, we should also filter out `dotComID == 0` blogs? But I'm not sure if such blog exists...

Reply:
>  we should also filter out `dotComID == 0` blogs? But I'm not sure if such blog exists...

Oh, right. 🤔 I just assumed that `0` was the fallback value, but I guess it's true: There's nothing preventing the value 0 from existing in the DB, however unlikely that might seem. Even if WordPress.com allows for ID = 0, there should be only _one_ such blog, right? Then again, if we didn't have issues with duplicated blogs, we wouldn't be writing a method to "de-duplicated"...

All this tells me I don't understand enough of what's going on under the hood and what the requirements are. Maybe it's best if I stop suggesting changes to things I don't understand 🙃

Reply:
> Even if WordPress.com allows for ID = 0, there should be only one such blog, right?

I surely hope there is no blog with a "0" id in the server, that'd be super weird 😄 . But on the app side though, I think it's still possible to have a `Blog` instance (like a self hosted site?) whose `dotComID` is 0.